google_gke_tenant/gke_service_account.tf

```diff
@@ -0,0 +1,33 @@
+resource "google_service_account" "gke-account" {
+  account_id  = "gke-${var.environment}"
+  description = "GKE deployments in the ${var.environment} environment"
+  project     = var.project_id
+}
+
+resource "google_service_account_iam_binding" "workload-identity-for-gke" {
+  service_account_id = google_service_account.gke-account.name
+  role               = "roles/iam.workloadIdentityUser"
+  members = [
+    "serviceAccount:${var.cluster_project_id}.svc.id.goog[${var.namespace}/external-secrets]"
+  ]
+}
+
+# permissions for use with External Secrets Operator in GKE
+#
+resource "google_project_iam_member" "sa-role-token-creator" {
+  project = var.project_id
+  role    = "roles/iam.serviceAccountTokenCreator"
```

Comment:
Is this role required? In [the docs](https://external-secrets.io/v0.5.8/provider-google-secrets-manager/#using-pod-based-workload-identity) it says:

> You need to give the Google service account the `roles/iam.serviceAccountTokenCreator` role so it can generate a service account token for you (not necessary in the Pod-based Workload Identity bellow)

If you wanted to keep this role for testing while you set it up, could you add a `FIXME` type comment so we know to come back to this?

Reply:
TBH, I didn't really understand the "Using Pod-based Workload Identity" section, so I went with the "Using Service Accounts directly" instructions above it.
Reading it again, it seems like maybe that'd require a container for ESO in every pod? Or only one service account for all namespaces? I don't get it. :(

Reply:
added FIXME